addons/account/models/account_invoice.py

```diff
@@ -1720,7 +1720,7 @@ def _get_invoice_line_name_from_product(self):
         self.ensure_one()
         if not self.product_id:
             return ''
-
+        type = self.invoice_id.type
```

Comment:
I prefer to not overwrite `type` variable, so please put the value inline in the if

Reply:
And you say that when the above method also have the same problem, and also all the file contains the same problem in many methods. IMO, doing this change you propose just for this method is just inconsiderable.

Reply:
In the future, I will do another PR to just propose the rename of those <code>type</code> variables in all odoo 😉

Reply:
It's not inconsiderable. It's just that you are touching THIS method, so it's the time for doing it properly. The rest should be changed of course in future PRs. Or are you going to continue introducing more bad things because the rest is done that way?

Reply:
done